packages/app/src/specs/SpecsListBanners.vue

```diff
@@ -254,6 +263,11 @@ watch(
     showConnectBanner.value = !hasBannerBeenDismissed(BannerIds.ACI_082022_CONNECT_PROJECT) && isLoggedIn && isMemberOfOrganization && !isProjectConnected && hasFourDaysOfCypressUse
     showCreateOrganizationBanner.value = !hasBannerBeenDismissed(BannerIds.ACI_082022_CREATE_ORG) && isLoggedIn && isOrganizationLoaded && !isMemberOfOrganization && hasFourDaysOfCypressUse
     showLoginBanner.value = !hasBannerBeenDismissed(BannerIds.ACI_082022_LOGIN) && !isLoggedIn && hasFourDaysOfCypressUse
+
```

Comment:
Not saying we do it in this PR, but this watcher is so huge and complex - I think if we get any more complex we should move this into it's own function somewhere else, and potentially test it in isolation (it could just be boring old JS function, not coupled to the UI, since 90% of it is just "if X then Y else Z". 

Interested in your thoughts here - fat components vs thin components kind of thing... more of a philosophical discussion a real comment or change request.

Reply:
Agreed, this is a monster. I know that the work Mark is doing will move a lot of this into a reusable wrapper component so I think that will help clean this up (or at least split the concerns into multiple smaller chunks). If that weren't the case I would lean towards building a couple composables to make some of the common checks this is doing a bit more reusable (`useIsLoggedIn`, `useIsOrgMember`, etc), and I'm sure most of the composable logic could be structured to test as pure boring JS